ext/ffi/lib.rs

```diff
@@ -1834,17 +1852,52 @@ where
   Ok(result)
 }
 
-#[op]
-fn op_ffi_unsafe_callback_ref(state: &mut deno_core::OpState, inc_dec: bool) {
-  check_unstable(state, "Deno.dlopen");
-  let ffi_state = state.borrow_mut::<FfiState>();
-  if inc_dec {
-    ffi_state.active_refed_functions += 1;
-  } else {
-    ffi_state.active_refed_functions -= 1;
+impl Future for CallbackInfo {
+  type Output = ();
+  fn poll(
+    self: Pin<&mut Self>,
+    cx: &mut std::task::Context<'_>,
+  ) -> std::task::Poll<Self::Output> {
+    // Always replace the waker to make sure it's bound to the proper Future.
+    self.waker.lock().unwrap().replace(cx.waker().clone());
+    // The future for the CallbackInfo never resolves: It can only be canceled.
+    Poll::Pending
   }
 }
 
+#[op]
+fn op_ffi_unsafe_callback_ref(
+  state: &mut deno_core::OpState,
+  rid: ResourceId,
+) -> Result<impl Future<Output = Result<(), AnyError>>, AnyError> {
```

Comment:
There's something odd going on with this op's hybridity: Even if the resource is already closed and the getting by rid from the resource table fails, it will still get rejected in a promise instead of being thrown synchronously. @littledivy Do you have any clue what might be wrong with this?

Reply:
I misremembered: The op macro purposefully (perhaps unnecessarily) handles even the synchronous result through the promise.